aten/src/ATen/templates/RegisterBackendSelect.cpp

```diff
@@ -15,8 +15,32 @@ namespace {
 
 ${backend_select_method_definitions}
 
+bool is_pinned(const Tensor& self, c10::optional<at::Device> device) {
+  // Only CPU tensors can be pinned
+  if (!self.is_cpu()) {
+    return false;
```

Comment:
Kinda confused by this. Won't this mean that if you call `is_pinned(my_tensor, kCuda)`, we'll just short-circuit here and return false instead of dispatching into `is_pinned_cuda`?

Also, if you want any short-circuiting here for not-allowed devices, isn't that redundant to the `is_pinned_default` kernel?

Reply:
oh, do we need this because of static dispatch? We want `is_pinned(_, kCPU)` to return false, not throw an error. Although another option would be to register `is_pinned_default` to both CPU and CompositeExplicitAutograd - not sure if that's better though.

Reply:
Look carefully: the test is on `self.is_cpu()`, not `device`. If you pass a CUDA tensor to is pinned, we always return false, because the pinning concept is taking CPU memory and putting it in a place where the CUDA driver can read it out directly. So in your example, so long as `my_tensor` is CPU, we will not short circuit here.

Reply:
bleh yeah, that makes sense. thanks!